From 5093dbc5f9f1d0414f6d25bf9969084831332d36 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:10:32 -0400
Subject: [usb] Revert: pl2303: simplify the else-if contruct for type_1 chips in pl2303_startup()

Message-id: <1410887463-20674-561-git-send-email-dzickus@redhat.com>
Patchwork-id: 95481
O-Subject: [RHEL7 PATCH 560/591] Revert "pl2303: simplify the else-if contruct for type_1 chips in pl2303_startup()"
Bugzilla: 1110939
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1110939

commit 281393ad0bcfc309434d2bff38abc15805c2cbc4
Author: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
Date:   Fri Nov 1 09:18:10 2013 -0700

    Revert "pl2303: simplify the else-if contruct for type_1 chips in pl2303_startup()"

    This reverts commit 73b583af597542329e6adae44524da6f27afed62.

    Revert all of the pl2303 changes that went into 3.12-rc1 and -rc2 as
    they cause regressions on some versions of the chip.  This will all be
    revisited for later kernel versions when we can figure out how to handle
    this in a way that does not break working devices.

    Reported-by: Mika Westerberg <mika.westerberg@linux.intel.com>
    Cc: Frank Schafer <fschaefer.oss@googlemail.com>
    Acked-by: Johan Hovold <jhovold@gmail.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/usb/serial/pl2303.c b/drivers/usb/serial/pl2303.c
index d4ca876..b4cfb86 100644
--- a/drivers/usb/serial/pl2303.c
+++ b/drivers/usb/serial/pl2303.c
@@ -196,8 +196,9 @@ static int pl2303_startup(struct usb_serial *serial)
   type = type_0;
  else if (serial->dev->descriptor.bMaxPacketSize0 == 0x40)
   type = HX;
- else if (serial->dev->descriptor.bDeviceClass == 0x00
-   || serial->dev->descriptor.bDeviceClass == 0xFF)
+ else if (serial->dev->descriptor.bDeviceClass == 0x00)
+  type = type_1;
+ else if (serial->dev->descriptor.bDeviceClass == 0xFF)
   type = type_1;
  dev_dbg(&serial->interface->dev, "device type: %d\n", type);
 
-- 
1.7.1